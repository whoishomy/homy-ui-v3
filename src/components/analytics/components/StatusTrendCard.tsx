'use client';

import React, { FC } from 'react';
import { PieChart, Pie, Cell } from 'recharts';
import { ArrowUpRightIcon, ArrowDownRightIcon } from '@heroicons/react/24/solid';
import { cn } from '@/lib/utils';
import { Info, TrendingUp, TrendingDown } from 'lucide-react';
import { Button } from '@/components/ui/Button';

export type StatusTrendType = 'info' | 'success' | 'warning' | 'error';

type StatusTrendCardProps = {
  title: string;
  score: number;
  change?: number;
  trendLabel?: string;
  className?: string;
  'aria-label'?: string;
  type?: StatusTrendType;
  formatValue?: (score: number) => string;
  aiRecommendation?: string;
  recommendationPriority?: 'low' | 'medium' | 'high';
  actionable?: boolean;
  onActionClick?: () => void;
};

const TYPE_STYLES: Record<StatusTrendType, string> = {
  info: 'text-blue-500 dark:text-blue-400',
  success: 'text-green-500 dark:text-green-400',
  warning: 'text-yellow-500 dark:text-yellow-400',
  error: 'text-red-500 dark:text-red-400',
};

const PRIORITY_STYLES = {
  low: 'bg-gray-50 text-gray-600 dark:bg-gray-800 dark:text-gray-400',
  medium: 'bg-yellow-50 text-yellow-700 dark:bg-yellow-900/30 dark:text-yellow-400',
  high: 'bg-red-50 text-red-700 dark:bg-red-900/30 dark:text-red-400',
};

const COLORS = ['#4ade80', '#e5e7eb']; // green + gray

export const StatusTrendCard: FC<StatusTrendCardProps> = ({
  title,
  score,
  change,
  trendLabel,
  className,
  'aria-label': ariaLabel,
  type = 'info',
  formatValue,
  aiRecommendation,
  recommendationPriority = 'low',
  actionable,
  onActionClick,
}) => {
  const percentage = Math.min(100, Math.max(0, score));
  const isPositive = (change ?? 0) >= 0;
  const typeStyle = type ? TYPE_STYLES[type] : '';

  const data = [
    { name: 'score', value: percentage },
    { name: 'rest', value: 100 - percentage },
  ];

  const formattedValue = formatValue ? formatValue(score) : `${score}%`;
  const trendIconLabel = isPositive ? 'Artış gösteriyor' : 'Düşüş gösteriyor';

  return (
    <div
      className={cn(
        'rounded-2xl bg-white shadow-sm ring-1 ring-gray-200 p-4 w-full max-w-sm',
        'dark:bg-gray-800 dark:ring-gray-700 dark:shadow-none',
        className
      )}
      role="status"
      aria-label={ariaLabel || `${title}: ${formattedValue}`}
    >
      <div className="flex items-start justify-between mb-2">
        <h3 className="text-sm font-medium text-gray-900 dark:text-gray-100">{title}</h3>
        {change !== undefined && (
          <div
            className={cn(
              'flex items-center text-xs font-medium',
              isPositive ? 'text-green-600' : 'text-red-600'
            )}
            role="text"
            aria-label={`${Math.abs(change)}% ${trendIconLabel}`}
          >
            {isPositive ? (
              <TrendingUp className="w-4 h-4 mr-1" role="img" aria-label={trendIconLabel} />
            ) : (
              <TrendingDown className="w-4 h-4 mr-1" role="img" aria-label={trendIconLabel} />
            )}
            {Math.abs(change)}%
          </div>
        )}
      </div>

      <div className="flex items-center space-x-4">
        <div className="relative w-16 h-16 flex items-center justify-center">
          <PieChart width={64} height={64}>
            <Pie
              data={data}
              cx={32}
              cy={32}
              innerRadius={22}
              outerRadius={30}
              startAngle={90}
              endAngle={-270}
              dataKey="value"
              stroke="none"
            >
              {data.map((entry, index) => (
                <Cell
                  key={`cell-${index}`}
                  fill={COLORS[index % COLORS.length]}
                  className={cn(
                    index === 0 ? typeStyle : 'text-gray-200 dark:text-gray-600',
                    'fill-current'
                  )}
                />
              ))}
            </Pie>
          </PieChart>
          <span className={cn('absolute text-sm font-medium', typeStyle)}>{formattedValue}</span>
        </div>

        <div className="flex-1 min-w-0">
          {trendLabel && (
            <p className="text-xs text-gray-500 dark:text-gray-400 mb-1">{trendLabel}</p>
          )}
          {aiRecommendation && (
            <div
              className={cn('mt-2 p-2 rounded-lg text-xs', PRIORITY_STYLES[recommendationPriority])}
            >
              <div className="flex items-start space-x-2">
                <Info className="w-4 h-4 mt-0.5 flex-shrink-0" aria-hidden="true" />
                <p>{aiRecommendation}</p>
              </div>
              {actionable && (
                <Button
                  className="mt-2 w-full justify-center hover:bg-white/10"
                  onClick={onActionClick}
                  aria-label={`${title} için önerileri gör`}
                >
                  Önerileri Gör
                </Button>
              )}
            </div>
          )}
        </div>
      </div>
    </div>
  );
};
