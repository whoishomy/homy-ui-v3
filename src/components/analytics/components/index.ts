export * from './InsightCard';
export * from './InsightList';
export * from './GoalCompletionRadar';
export * from './HealthTrendCards';
export * from './CategoryFilter';
export * from './TimeRangeSelector';
export * from './InsightsSummary';
export * from './MonthlyAdherenceCalendar';
export { InsightCard } from './InsightCard'; 