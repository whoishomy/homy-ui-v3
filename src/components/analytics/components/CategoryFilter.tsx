import React from 'react';
import { HealthCategory } from '@/types/analytics';

interface CategoryFilterProps {
  selected: Array<keyof typeof HealthCategory>;
  onChange: (categories: Array<keyof typeof HealthCategory>) => void;
}

export const CategoryFilter: React.FC<CategoryFilterProps> = ({
  selected,
  onChange,
}) => {
  const handleToggle = (category: keyof typeof HealthCategory) => {
    if (selected.includes(category)) {
      onChange(selected.filter((c) => c !== category));
    } else {
      onChange([...selected, category]);
    }
  };

  return (
    <div className="flex flex-wrap gap-2">
      {Object.entries(HealthCategory).map(([key, label]) => (
        <button
          key={key}
          onClick={() => handleToggle(key as keyof typeof HealthCategory)}
          className={`inline-flex items-center px-3 py-1.5 rounded-full text-sm font-medium transition-colors ${
            selected.includes(key as keyof typeof HealthCategory)
              ? 'bg-green-100 text-green-800'
              : 'bg-gray-100 text-gray-700 hover:bg-gray-200'
          }`}
        >
          {label}
        </button>
      ))}
    </div>
  );
}; 