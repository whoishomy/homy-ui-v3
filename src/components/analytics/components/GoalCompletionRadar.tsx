import React from 'react';
import {
  Radar,
  RadarChart,
  PolarGrid,
  PolarAngleAxis,
  PolarRadiusAxis,
  ResponsiveContainer,
  Legend,
} from 'recharts';
import { HealthCategory } from '@/types/analytics';

interface GoalCompletionRadarProps {
  data: {
    category: keyof typeof HealthCategory;
    completion: number;
    target: number;
  }[];
}

export const GoalCompletionRadar: React.FC<GoalCompletionRadarProps> = ({
  data,
}) => {
  const formattedData = data.map((item) => ({
    category: HealthCategory[item.category],
    completion: item.completion,
    target: item.target,
  }));

  return (
    <div className="w-full h-[400px]">
      <ResponsiveContainer width="100%" height="100%">
        <RadarChart cx="50%" cy="50%" outerRadius="80%" data={formattedData}>
          <PolarGrid stroke="#E5E7EB" />
          <PolarAngleAxis
            dataKey="category"
            tick={{ fill: '#6B7280', fontSize: 12 }}
          />
          <PolarRadiusAxis
            angle={30}
            domain={[0, 100]}
            tick={{ fill: '#6B7280', fontSize: 10 }}
          />
          <Radar
            name="Tamamlama"
            dataKey="completion"
            stroke="#10B981"
            fill="#10B981"
            fillOpacity={0.6}
          />
          <Radar
            name="Hedef"
            dataKey="target"
            stroke="#6B7280"
            fill="#6B7280"
            fillOpacity={0.3}
          />
          <Legend
            wrapperStyle={{
              paddingTop: '2rem',
            }}
          />
        </RadarChart>
      </ResponsiveContainer>
    </div>
  );
}; 