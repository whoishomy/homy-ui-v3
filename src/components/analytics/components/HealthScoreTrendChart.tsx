import React from 'react';
import {
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend,
  ResponsiveContainer,
} from 'recharts';
import { format } from 'date-fns';
import { tr } from 'date-fns/locale';
import { TrendData } from '@/types/analytics';

interface HealthScoreTrendChartProps {
  data: TrendData[];
}

export const HealthScoreTrendChart: React.FC<HealthScoreTrendChartProps> = ({
  data,
}) => {
  const formattedData = data.map((item) => ({
    ...item,
    date: format(new Date(item.date), 'dd MMM', { locale: tr }),
  }));

  return (
    <div className="w-full h-[300px]">
      <ResponsiveContainer width="100%" height="100%">
        <LineChart
          data={formattedData}
          margin={{
            top: 5,
            right: 30,
            left: 20,
            bottom: 5,
          }}
        >
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis
            dataKey="date"
            tick={{ fill: '#6B7280' }}
            tickLine={{ stroke: '#6B7280' }}
          />
          <YAxis
            tick={{ fill: '#6B7280' }}
            tickLine={{ stroke: '#6B7280' }}
            domain={[0, 100]}
          />
          <Tooltip
            contentStyle={{
              backgroundColor: '#ffffff',
              border: '1px solid #e5e7eb',
              borderRadius: '0.375rem',
            }}
            labelStyle={{ color: '#111827' }}
            itemStyle={{ color: '#6B7280' }}
          />
          <Legend
            wrapperStyle={{
              paddingTop: '1rem',
            }}
          />
          <Line
            type="monotone"
            dataKey="value"
            name="Sağlık Skoru"
            stroke="#10B981"
            strokeWidth={2}
            dot={{
              fill: '#10B981',
              stroke: '#ffffff',
              strokeWidth: 2,
              r: 4,
            }}
            activeDot={{
              fill: '#10B981',
              stroke: '#ffffff',
              strokeWidth: 2,
              r: 6,
            }}
          />
          {data[0]?.target && (
            <Line
              type="monotone"
              dataKey="target"
              name="Hedef"
              stroke="#6B7280"
              strokeDasharray="5 5"
              strokeWidth={2}
              dot={false}
            />
          )}
        </LineChart>
      </ResponsiveContainer>
    </div>
  );
}; 