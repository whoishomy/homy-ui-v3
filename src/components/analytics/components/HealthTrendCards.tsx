import React from 'react';
import { Activity, TrendingUp, TrendingDown } from 'lucide-react';
import { AnalyticsSummary } from '@/types/analytics';

interface HealthTrendCardsProps {
  summary: AnalyticsSummary | null;
}

export const HealthTrendCards: React.FC<HealthTrendCardsProps> = ({ summary }) => {
  if (!summary) return null;

  const overallTrend = summary.trends[summary.trends.length - 1]?.value || 0;
  const previousTrend = summary.trends[summary.trends.length - 2]?.value || 0;
  const trendDiff = overallTrend - previousTrend;
  const trendPercentage = ((trendDiff / previousTrend) * 100).toFixed(1);

  const cards = [
    {
      title: 'Genel Sağlık Skoru',
      value: `${Math.round(overallTrend)}%`,
      trend: {
        value: trendPercentage,
        direction: trendDiff >= 0 ? 'up' : 'down',
      },
      icon: Activity,
    },
    {
      title: 'Ortalama Uyum',
      value: `${Math.round(summary.adherence.overall)}%`,
      trend: {
        value: '0.0',
        direction: 'up',
      },
      icon: TrendingUp,
    },
    {
      title: 'Hedef Tamamlama',
      value: `${Math.round(
        summary.goalCompletion.reduce((acc, curr) => acc + curr.completion, 0) /
          summary.goalCompletion.length
      )}%`,
      trend: {
        value: '0.0',
        direction: 'up',
      },
      icon: TrendingDown,
    },
  ];

  return (
    <div className="grid grid-cols-1 md:grid-cols-3 gap-4">
      {cards.map((card) => {
        const Icon = card.icon;
        const isPositive = card.trend.direction === 'up';

        return (
          <div
            key={card.title}
            className="bg-white rounded-lg shadow p-6 flex items-start space-x-4"
          >
            <div
              className={`p-3 rounded-full ${
                isPositive ? 'bg-green-100' : 'bg-red-100'
              }`}
            >
              <Icon
                className={`w-6 h-6 ${
                  isPositive ? 'text-green-600' : 'text-red-600'
                }`}
              />
            </div>
            <div>
              <p className="text-sm font-medium text-gray-600">{card.title}</p>
              <p className="text-2xl font-semibold text-gray-900">{card.value}</p>
              <div className="mt-1 flex items-center">
                {card.trend.direction === 'up' ? (
                  <TrendingUp className="w-4 h-4 text-green-500" />
                ) : (
                  <TrendingDown className="w-4 h-4 text-red-500" />
                )}
                <span
                  className={`ml-1 text-sm font-medium ${
                    isPositive ? 'text-green-600' : 'text-red-600'
                  }`}
                >
                  {card.trend.value}%
                </span>
                <span className="ml-1 text-sm text-gray-500">
                  geçen haftaya göre
                </span>
              </div>
            </div>
          </div>
        );
      })}
    </div>
  );
}; 