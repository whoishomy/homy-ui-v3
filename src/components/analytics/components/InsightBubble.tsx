'use client';

import React from 'react';
import { clsx } from 'clsx';
import { motion } from 'framer-motion';
import { Lightbulb, AlertTriangle, CheckCircle2, Info } from 'lucide-react';
import { Tooltip } from '@/components/ui/Tooltip';
import { InsightCategory } from '@/types/analytics';
import { useInsightCategoryDisplay } from '@/hooks/useInsightCategoryDisplay';
import { useTranslation } from 'next-i18next';

type InsightType = 'info' | 'success' | 'warning' | 'critical';

interface InsightBubbleProps {
  id?: string;
  message: string;
  type?: InsightType;
  priority?: 'low' | 'medium' | 'high';
  category?: InsightCategory;
  className?: string;
  tooltipContent?: React.ReactNode;
  onAction?: (action: string) => void;
}

const ICONS: Record<InsightType, React.ReactNode> = {
  info: <Info className="w-5 h-5 text-blue-500 dark:text-blue-400" />,
  success: <CheckCircle2 className="w-5 h-5 text-green-500 dark:text-green-400" />,
  warning: <AlertTriangle className="w-5 h-5 text-yellow-500 dark:text-yellow-400" />,
  critical: <AlertTriangle className="w-5 h-5 text-red-600 dark:text-red-500" />,
};

const bubbleVariants = {
  initial: {
    opacity: 0,
    y: 20,
    scale: 0.95,
  },
  animate: {
    opacity: 1,
    y: 0,
    scale: 1,
    transition: {
      duration: 0.3,
      ease: 'easeOut',
    },
  },
  hover: {
    scale: 1.02,
    transition: {
      duration: 0.2,
    },
  },
  tap: {
    scale: 0.98,
  },
};

export const InsightBubble: React.FC<InsightBubbleProps> = ({
  message,
  type = 'info',
  priority = 'medium',
  category,
  className = '',
  tooltipContent,
}) => {
  const { t } = useTranslation('analytics');
  const { getDisplay } = useInsightCategoryDisplay();
  const showTooltip = priority === 'high' && (type === 'critical' || type === 'warning');
  const defaultTooltip = t('insights.tooltips.default');

  const bubble = (
    <motion.div
      role="note"
      aria-live="polite"
      variants={bubbleVariants}
      initial="initial"
      animate="animate"
      whileHover="hover"
      whileTap="tap"
      className={clsx(
        'flex items-start gap-2 p-3 rounded-md shadow-sm border text-sm transition-colors cursor-default',
        {
          'bg-blue-50 text-blue-800 border-blue-100 dark:bg-blue-900/10 dark:text-blue-400 dark:border-blue-900/20':
            type === 'info',
          'bg-green-50 text-green-800 border-green-100 dark:bg-green-900/10 dark:text-green-400 dark:border-green-900/20':
            type === 'success',
          'bg-yellow-50 text-yellow-900 border-yellow-200 dark:bg-yellow-900/10 dark:text-yellow-400 dark:border-yellow-900/20':
            type === 'warning',
          'bg-red-50 text-red-800 border-red-200 dark:bg-red-900/10 dark:text-red-400 dark:border-red-900/20':
            type === 'critical',
        },
        className
      )}
    >
      <motion.div
        className="pt-0.5"
        initial={{ rotate: -10, scale: 0.5 }}
        animate={{ rotate: 0, scale: 1 }}
        transition={{ duration: 0.3, delay: 0.1 }}
      >
        {ICONS[type]}
      </motion.div>
      <div className="flex-1">
        <p>{message}</p>
        {category && <span className="text-xs mt-1 opacity-75">{getDisplay(category)}</span>}
      </div>
      {priority === 'high' && (
        <motion.span
          className="ml-2 text-xs font-semibold text-red-600 dark:text-red-500"
          initial={{ opacity: 0, x: -10 }}
          animate={{ opacity: 1, x: 0 }}
          transition={{ duration: 0.3, delay: 0.2 }}
        >
          {t('insights.priority')}
        </motion.span>
      )}
    </motion.div>
  );

  if (showTooltip) {
    return <Tooltip content={tooltipContent || defaultTooltip}>{bubble}</Tooltip>;
  }

  return bubble;
};
