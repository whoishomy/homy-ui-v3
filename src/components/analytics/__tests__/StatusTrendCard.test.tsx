import { render, screen } from '@testing-library/react';
import { StatusTrendCard } from '../components/StatusTrendCard';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);

describe('StatusTrendCard', () => {
  const defaultProps = {
    title: 'Test Metric',
    score: 85.5,
    change: 5.2,
    trendLabel: 'son 24 saat',
  };

  describe('Rendering', () => {
    it('renders basic card with title and score', () => {
      render(<StatusTrendCard {...defaultProps} />);

      expect(screen.getByText('Test Metric')).toBeInTheDocument();
      expect(screen.getByText('85.5%')).toBeInTheDocument();
    });

    it('applies correct type-based styles', () => {
      const { rerender } = render(<StatusTrendCard {...defaultProps} type="success" />);
      expect(screen.getByText('85.5%')).toHaveClass('text-green-500');

      rerender(<StatusTrendCard {...defaultProps} type="error" />);
      expect(screen.getByText('85.5%')).toHaveClass('text-red-500');

      rerender(<StatusTrendCard {...defaultProps} type="warning" />);
      expect(screen.getByText('85.5%')).toHaveClass('text-yellow-500');

      rerender(<StatusTrendCard {...defaultProps} type="info" />);
      expect(screen.getByText('85.5%')).toHaveClass('text-blue-500');
    });

    it('formats value using custom formatter', () => {
      render(<StatusTrendCard {...defaultProps} formatValue={(v) => `${v.toFixed(0)}ms`} />);

      expect(screen.getByText('86ms')).toBeInTheDocument();
    });

    it('shows trend indicator and label when provided', () => {
      render(<StatusTrendCard {...defaultProps} />);

      const trendValue = screen.getByText('5.2%');
      const trendLabel = screen.getByText('son 24 saat');

      expect(trendValue).toBeInTheDocument();
      expect(trendLabel).toBeInTheDocument();
    });

    it('handles negative change values', () => {
      render(<StatusTrendCard {...defaultProps} change={-3.5} />);

      const trendValue = screen.getByText('3.5%');
      expect(trendValue).toHaveClass('text-red-600');
    });
  });

  describe('Accessibility', () => {
    it('has no accessibility violations', async () => {
      const { container } = render(<StatusTrendCard {...defaultProps} />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });

    it('uses correct ARIA labels', () => {
      render(<StatusTrendCard {...defaultProps} />);

      const card = screen.getByRole('status');
      expect(card).toHaveAttribute('aria-label', 'Test Metric: 85.5%');
    });

    it('uses custom ARIA label when provided', () => {
      render(<StatusTrendCard {...defaultProps} aria-label="Custom metric status" />);

      const card = screen.getByRole('status');
      expect(card).toHaveAttribute('aria-label', 'Custom metric status');
    });

    it('provides descriptive labels for trend icons', () => {
      const { rerender } = render(<StatusTrendCard {...defaultProps} change={5.2} />);

      let trendIcon = screen.getByRole('img');
      expect(trendIcon).toHaveAttribute('aria-label', 'Artış gösteriyor');

      rerender(<StatusTrendCard {...defaultProps} change={-3.5} />);
      trendIcon = screen.getByRole('img');
      expect(trendIcon).toHaveAttribute('aria-label', 'Düşüş gösteriyor');
    });

    it('provides combined trend information', () => {
      const { rerender } = render(<StatusTrendCard {...defaultProps} change={5.2} />);

      let trendContainer = screen.getByRole('text');
      expect(trendContainer).toHaveAttribute('aria-label', '5.2% Artış gösteriyor');

      rerender(<StatusTrendCard {...defaultProps} change={-3.5} />);
      trendContainer = screen.getByRole('text');
      expect(trendContainer).toHaveAttribute('aria-label', '3.5% Düşüş gösteriyor');
    });

    it('marks info icon as decorative', () => {
      render(<StatusTrendCard {...defaultProps} aiRecommendation="Test recommendation" />);
      const infoIcon = screen.getByRole('presentation', { hidden: true });
      expect(infoIcon).toHaveAttribute('aria-hidden', 'true');
    });
  });

  describe('Dark Mode', () => {
    it('applies dark mode classes', () => {
      const { container } = render(<StatusTrendCard {...defaultProps} />);

      expect(container.firstChild).toHaveClass('dark:bg-gray-800');
      expect(container.firstChild).toHaveClass('dark:ring-gray-700');
    });

    it('uses correct dark mode text colors', () => {
      render(<StatusTrendCard {...defaultProps} type="success" />);

      expect(screen.getByText('Test Metric')).toHaveClass('dark:text-gray-100');
      expect(screen.getByText('85.5%')).toHaveClass('dark:text-green-400');
    });
  });

  describe('Edge Cases', () => {
    it('clamps score to 0-100 range', () => {
      const { rerender } = render(<StatusTrendCard {...defaultProps} score={150} />);
      expect(screen.getByText('100%')).toBeInTheDocument();

      rerender(<StatusTrendCard {...defaultProps} score={-20} />);
      expect(screen.getByText('0%')).toBeInTheDocument();
    });

    it('handles undefined change and trendLabel', () => {
      render(<StatusTrendCard title="Test Metric" score={85} />);

      expect(screen.queryByRole('img')).not.toBeInTheDocument();
      expect(screen.queryByText('son 24 saat')).not.toBeInTheDocument();
    });

    it('handles zero score value', () => {
      render(<StatusTrendCard {...defaultProps} score={0} />);
      expect(screen.getByText('0%')).toBeInTheDocument();
    });
  });

  describe('Visual Regression', () => {
    const types: Array<'info' | 'success' | 'warning' | 'error'> = [
      'info',
      'success',
      'warning',
      'error',
    ];

    types.forEach((type) => {
      it(`matches ${type} type snapshot`, () => {
        const { container } = render(<StatusTrendCard {...defaultProps} type={type} />);
        expect(container).toMatchSnapshot();
      });
    });

    it('matches snapshot with custom formatting', () => {
      const { container } = render(
        <StatusTrendCard {...defaultProps} formatValue={(v) => `${v.toFixed(1)}ms`} />
      );
      expect(container).toMatchSnapshot();
    });
  });
});
