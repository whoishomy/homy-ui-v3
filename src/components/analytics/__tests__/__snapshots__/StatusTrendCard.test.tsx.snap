// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusTrendCard Visual Regression matches error type snapshot 1`] = `
<div>
  <div
    aria-label="Test Metric: 85.5%"
    class="rounded-2xl bg-white shadow-sm ring-1 ring-gray-200 p-4 w-full max-w-sm dark:bg-gray-800 dark:ring-gray-700 dark:shadow-none"
    role="status"
  >
    <div
      class="flex items-start justify-between mb-2"
    >
      <h3
        class="text-sm font-medium text-gray-900 dark:text-gray-100"
      >
        Test Metric
      </h3>
      <div
        aria-label="5.2% Artış gösteriyor"
        class="flex items-center text-xs font-medium text-green-600"
        role="text"
      >
        <svg
          aria-label="Artış gösteriyor"
          class="lucide lucide-trending-up w-4 h-4 mr-1"
          fill="none"
          height="24"
          role="img"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="22 7 13.5 15.5 8.5 10.5 2 17"
          />
          <polyline
            points="16 7 22 7 22 13"
          />
        </svg>
        5.2
        %
      </div>
    </div>
    <div
      class="flex items-center space-x-4"
    >
      <div
        class="relative w-16 h-16 flex items-center justify-center"
      >
        <span
          class="absolute text-sm font-medium text-red-500 dark:text-red-400"
        >
          85.5%
        </span>
      </div>
      <div
        class="flex-1 min-w-0"
      >
        <p
          class="text-xs text-gray-500 dark:text-gray-400 mb-1"
        >
          son 24 saat
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`StatusTrendCard Visual Regression matches info type snapshot 1`] = `
<div>
  <div
    aria-label="Test Metric: 85.5%"
    class="rounded-2xl bg-white shadow-sm ring-1 ring-gray-200 p-4 w-full max-w-sm dark:bg-gray-800 dark:ring-gray-700 dark:shadow-none"
    role="status"
  >
    <div
      class="flex items-start justify-between mb-2"
    >
      <h3
        class="text-sm font-medium text-gray-900 dark:text-gray-100"
      >
        Test Metric
      </h3>
      <div
        aria-label="5.2% Artış gösteriyor"
        class="flex items-center text-xs font-medium text-green-600"
        role="text"
      >
        <svg
          aria-label="Artış gösteriyor"
          class="lucide lucide-trending-up w-4 h-4 mr-1"
          fill="none"
          height="24"
          role="img"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="22 7 13.5 15.5 8.5 10.5 2 17"
          />
          <polyline
            points="16 7 22 7 22 13"
          />
        </svg>
        5.2
        %
      </div>
    </div>
    <div
      class="flex items-center space-x-4"
    >
      <div
        class="relative w-16 h-16 flex items-center justify-center"
      >
        <span
          class="absolute text-sm font-medium text-blue-500 dark:text-blue-400"
        >
          85.5%
        </span>
      </div>
      <div
        class="flex-1 min-w-0"
      >
        <p
          class="text-xs text-gray-500 dark:text-gray-400 mb-1"
        >
          son 24 saat
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`StatusTrendCard Visual Regression matches snapshot with custom formatting 1`] = `
<div>
  <div
    aria-label="Test Metric: 85.5ms"
    class="rounded-2xl bg-white shadow-sm ring-1 ring-gray-200 p-4 w-full max-w-sm dark:bg-gray-800 dark:ring-gray-700 dark:shadow-none"
    role="status"
  >
    <div
      class="flex items-start justify-between mb-2"
    >
      <h3
        class="text-sm font-medium text-gray-900 dark:text-gray-100"
      >
        Test Metric
      </h3>
      <div
        aria-label="5.2% Artış gösteriyor"
        class="flex items-center text-xs font-medium text-green-600"
        role="text"
      >
        <svg
          aria-label="Artış gösteriyor"
          class="lucide lucide-trending-up w-4 h-4 mr-1"
          fill="none"
          height="24"
          role="img"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="22 7 13.5 15.5 8.5 10.5 2 17"
          />
          <polyline
            points="16 7 22 7 22 13"
          />
        </svg>
        5.2
        %
      </div>
    </div>
    <div
      class="flex items-center space-x-4"
    >
      <div
        class="relative w-16 h-16 flex items-center justify-center"
      >
        <span
          class="absolute text-sm font-medium text-blue-500 dark:text-blue-400"
        >
          85.5ms
        </span>
      </div>
      <div
        class="flex-1 min-w-0"
      >
        <p
          class="text-xs text-gray-500 dark:text-gray-400 mb-1"
        >
          son 24 saat
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`StatusTrendCard Visual Regression matches success type snapshot 1`] = `
<div>
  <div
    aria-label="Test Metric: 85.5%"
    class="rounded-2xl bg-white shadow-sm ring-1 ring-gray-200 p-4 w-full max-w-sm dark:bg-gray-800 dark:ring-gray-700 dark:shadow-none"
    role="status"
  >
    <div
      class="flex items-start justify-between mb-2"
    >
      <h3
        class="text-sm font-medium text-gray-900 dark:text-gray-100"
      >
        Test Metric
      </h3>
      <div
        aria-label="5.2% Artış gösteriyor"
        class="flex items-center text-xs font-medium text-green-600"
        role="text"
      >
        <svg
          aria-label="Artış gösteriyor"
          class="lucide lucide-trending-up w-4 h-4 mr-1"
          fill="none"
          height="24"
          role="img"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="22 7 13.5 15.5 8.5 10.5 2 17"
          />
          <polyline
            points="16 7 22 7 22 13"
          />
        </svg>
        5.2
        %
      </div>
    </div>
    <div
      class="flex items-center space-x-4"
    >
      <div
        class="relative w-16 h-16 flex items-center justify-center"
      >
        <span
          class="absolute text-sm font-medium text-green-500 dark:text-green-400"
        >
          85.5%
        </span>
      </div>
      <div
        class="flex-1 min-w-0"
      >
        <p
          class="text-xs text-gray-500 dark:text-gray-400 mb-1"
        >
          son 24 saat
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`StatusTrendCard Visual Regression matches warning type snapshot 1`] = `
<div>
  <div
    aria-label="Test Metric: 85.5%"
    class="rounded-2xl bg-white shadow-sm ring-1 ring-gray-200 p-4 w-full max-w-sm dark:bg-gray-800 dark:ring-gray-700 dark:shadow-none"
    role="status"
  >
    <div
      class="flex items-start justify-between mb-2"
    >
      <h3
        class="text-sm font-medium text-gray-900 dark:text-gray-100"
      >
        Test Metric
      </h3>
      <div
        aria-label="5.2% Artış gösteriyor"
        class="flex items-center text-xs font-medium text-green-600"
        role="text"
      >
        <svg
          aria-label="Artış gösteriyor"
          class="lucide lucide-trending-up w-4 h-4 mr-1"
          fill="none"
          height="24"
          role="img"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="22 7 13.5 15.5 8.5 10.5 2 17"
          />
          <polyline
            points="16 7 22 7 22 13"
          />
        </svg>
        5.2
        %
      </div>
    </div>
    <div
      class="flex items-center space-x-4"
    >
      <div
        class="relative w-16 h-16 flex items-center justify-center"
      >
        <span
          class="absolute text-sm font-medium text-yellow-500 dark:text-yellow-400"
        >
          85.5%
        </span>
      </div>
      <div
        class="flex-1 min-w-0"
      >
        <p
          class="text-xs text-gray-500 dark:text-gray-400 mb-1"
        >
          son 24 saat
        </p>
      </div>
    </div>
  </div>
</div>
`;
